velox/functions/lib/MapFromEntries.cpp

```diff
@@ -141,8 +145,14 @@ class MapFromEntriesFunction : public exec::VectorFunction {
           // Check nulls in the top level row vector.
           const bool isMapEntryNull = decodedRowVector->isNullAt(offset + i);
           if (isMapEntryNull) {
-            // Set the sizes to 0 so that the final map vector generated is
-            // valid in case we are inside a try. The map vector needs to be
+            if constexpr (!throwForNull) {
+              // Spark: For nulls in the top level row vector, return null.
+              bits::setNull(mutableNulls, row);
```

Comment:
Maybe we don't need mention Spark and Presto in the comment again, because the function description already covers it.